.circleci/config.yml

```diff
@@ -207,6 +216,12 @@ commands:
             - vms/
             - wasi-sdk/
             - .jsvu/
+      - save_cache:
+          name: "Save Ccache cache"
+          paths:
+            - ~/.ccache
+          key: clang-{{ checksum "~/emsdk/clang_version.txt" }}
```

Comment:
So this will mean that each clang version has its own cache, so that cache effectively be invalidated on each llvm roll.

However, what about changes outside the llvm?  i.e. emscripten changes?   How should we hangle them?  How does ccache decided if the compiler itself has changed?  Has it hash the compile binary or something like that?  Or is it just the command line string?

Reply:
ccache is only caching the output of clang itself. So if we invalidate the cache each time clang changes, that solves the problem of deciding whether the compiler has changed.

Reply:
(and for emscripten changes: if clang's input changes, e.g. the sources of the libraries, then ccache handles that already with it usual hashing; and if something outside of that changes, then it doesn't affect the library build and we want a cache hit)

Reply:
Oh nice!

Reply:
I think the link time will massively dominate that tests TBH